HospitalSystem.Core/Scheduling/Scheduler/ScheduleService.cs

```diff
@@ -283,20 +96,57 @@ public bool IsDoctorAvailable(DateRange range, Doctor doctor)
         return true;
     }
 
-    public int CompareCheckupsByDoctorsName(Checkup checkup1, Checkup checkup2)
+    public void ScheduleCheckup(Checkup newCheckup)
     {
-        string name1 = _doctorService.GetById((ObjectId)checkup1.Doctor.Id).FirstName;
-        string name2 = _doctorService.GetById((ObjectId)checkup2.Doctor.Id).FirstName;
-        return String.Compare(name1, name2);
+        newCheckup.RoomLocation = GetAvailableRoom(newCheckup, RoomType.CHECKUP).Location;
+        _appointmentService.UpsertCheckup(newCheckup);
+    }
+    public void ScheduleOperation(Operation newOperation)
+    {
+        newOperation.RoomLocation = GetAvailableRoom(newOperation, RoomType.OPERATION).Location;
+        _appointmentService.UpsertOperation(newOperation);
     }
 
-    public int CompareCheckupsByDoctorsSpecialty(Checkup checkup1, Checkup checkup2)
+    public bool ScheduleCheckup(User _user, DateTime dateTime, string name, string surname)
+    {
+        Patient patient = _patientService.GetPatientByFullName(name,surname);
+        if (patient == null)
+        {
+            return false;
+        }
+        Doctor doctor = _doctorService.GetById((ObjectId)_user.Person.Id);
+        Checkup checkup = new Checkup(dateTime, new MongoDBRef("patients", patient.Id), new MongoDBRef("doctors", _user.Person.Id), "anamnesis:");
+        if (IsDoctorAvailable(checkup.DateRange, doctor))
+        {
+            ScheduleCheckup(checkup);
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+    public bool ScheduleOperation(User _user, DateTime dateTime, string name, string surname, TimeSpan duration)
     {
-        string specialty1 = _doctorService.GetById((ObjectId)checkup1.Doctor.Id).Specialty.ToString();
-        string specialty2 = _doctorService.GetById((ObjectId)checkup2.Doctor.Id).Specialty.ToString();
-        return String.Compare(specialty1, specialty2);
+        Patient patient = _patientService.GetPatientByFullName(name,surname);
+        if (patient == null)
+        {
+            return false;
+        }
+        Doctor doctor = _doctorService.GetById((ObjectId)_user.Person.Id);
+        Operation operation = new Operation(new DateRange(dateTime, dateTime.Add(duration)), new MongoDBRef("patients", patient.Id), new MongoDBRef("doctors", _user.Person.Id), "anamnesis:");
+        if (IsDoctorAvailable(operation.DateRange, doctor))
+        {
+            ScheduleOperation(operation);
+            return true;
+        }
+        else
+        {
+            return false;
+        }
     }
 
+    //TO-DO: refactor this method
     public List<Checkup> GetEarliestFreeCheckups(DateRange interval, Specialty speciality, int numberOfCheckups, User user)
```

Comment:
TODO is written without a dash inbetween. If you write it like this the extensions won't automatically pick it up